### PR TITLE
Added 'check for updates' button

### PR DESCRIPTION
**Pull request by vjanssens**
Added a 'check for updates' button that links to the latest release page of Cumulus on GitHub.
Not the full auto-update feature issue #39 asks for, but it's a start which I think is useful.

image

### CODE REVIEW

**Comment by MichielDeMey** [comment]
I'd love to see an integration with the GitHub API as well, we can grab the latest release tag using:
https://api.github.com/repos/gillesdemey/cumulus/releases/latest
and compare that with the current application version (from the `package.json`).

That way we can show a nice dialog message in case the user is up-to-date.

**Comment by vjanssens** [comment]
Thanks for pointing me in the right direction, my latest commit contains a call to the GitHub API and showing a dialog box, as seen below.

image

As you can see, I haven't been able to change the icon in the dialog because i'm not sure what image to use and the dialog does not display at all when I add `icon: __dirname + '/app/IconTemplate.png'` to mediaPlayer.js r.137. Maybe you could give it a try?

Another thing I'm not really happy about is mediaPlayer.js r.150. I would like to use the `openExternal()` method on the mediaPlayer class, but `this` is not bound to the mediaPlayer class anymore inside the promise. It works right now, so this would only be an improvement.

As for the version comparison, I've kept it as simple as possible. It just compares the tag from GitHub and the tag inside package.json and if they are equal, the app is up-to-date.

Can you let me know what you think? Thanks :)

File: app/js/components/mediaPlayer.js [hunk 1]
```diff
@@ -5,13 +5,15 @@ var Actions           = require('../actions/actionCreators')
...
 var _                 = require('lodash')
 
-var remote            = window.require('remote');
-var Menu              = remote.require('menu');
-var MenuItem          = remote.require('menu-item');
-var pjson             = remote.require('./package.json');
+var remote            = window.require('remote')
+var Menu              = remote.require('menu')
+var MenuItem          = remote.require('menu-item')
+var pjson             = remote.require('./package.json')
```
Comment (MichielDeMey) [inline app/js/components/mediaPlayer.js]: We actually have a `config` module that takes care of reading values from the `package.json`, to maintain consistency I would recommend using that one.
https://github.com/gillesdemey/Cumulus/blob/master/lib/config.js#L22

**Comment by vjanssens** [comment]
No problem, if there is anything else please let me know :+1:

File: app/js/utils/github.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+GitHub.prototype.getRepoUrl = function() {
+  return this._repo;
+}
+
+GitHub.prototype.checkForUpdates = function() {
+  return this.getLatestRelease().then(function(response) {
+    var tag = response.tag_name.replace(/v/g, '')
+
+    if(pjson.version == tag) {
+      return Promise.resolve('Cumulus is currently up-to-date with the most recent version.')
```
Comment (MichielDeMey) [inline app/js/utils/github.js]: I would shorten it to `'Cumulus is currently up-to-date.'`.

File: app/js/utils/github.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+      return Promise.reject('Cumulus is currently not up-to-date.')
```
Comment (MichielDeMey) [inline app/js/utils/github.js]: I would change this to a more informative message like
```
A new version of Cumulus is available.
Go to http://gillesdemey.github.io/Cumulus/ to download the latest version.
```

And hopefully we can create a clickable link out of the url. :)

**Comment by vjanssens** [comment]
Sorry for the delay, I finally had some time to clean up the last bits today.
I've changed the strings as per your request and added the Cumulus icon. The 'update available' dialog has a 'Download latest version' button which leads to https://github.com/gillesdemey/Cumulus/releases/latest.

cumuls-download-latest-version

**Comment by gillesdemey** [comment]
LGTM! 👍

**Pull request merged**